### PR TITLE
Cluster status now honored at the transport layer.

### PR DESCRIPTION
**Pull request by ashcrow**

### CODE REVIEW

File: src/commissaire/jobs/investigator.py [hunk 1]
```diff
@@ -132,7 +134,7 @@ def investigator(queue, config, run_once=False):
         oscmd = get_oscmd(host.os)
         try:
             result, facts = transport.bootstrap(
-                address, key_file, config, oscmd)
+                address, key_file, config, oscmd, store_manager.clone())
```
Comment (mbarnes) [inline src/commissaire/jobs/investigator.py]: I don't think you need to use `clone()` here.  The `store_manager` instance popped off the queue has already been cloned for the investigator subprocess.  Not hurting anything, it's just unnecessary.

File: src/commissaire/jobs/investigator.py [hunk 2]
```diff
@@ -146,31 +148,42 @@ def investigator(queue, config, run_once=False):
...
-                    '{0} has not been registered with the container manager. '
-                    'Checking again in 5 seconds...'.format(address))
-                sleep(5)
-        except:
-            _, exc_msg, _ = sys.exc_info()
-            logger.warn(
-                'Unable to finish bootstrap for {0} while associating with '
-                'the container manager: {1}'.format(address, exc_msg))
-            host.status = 'inactive'
+            cluster = util.cluster_for_host(address, store_manager.clone())
```
Comment (mbarnes) [inline src/commissaire/jobs/investigator.py]: Same here with the unnecessary `clone()`.

File: src/commissaire/transport/ansibleapi.py [hunk 3]
```diff
@@ -352,13 +355,23 @@ def bootstrap(self, ip, key_file, config, oscmd):
         :param config: Configuration information.
         :type config: commissaire.config.Config
         :param oscmd: OSCmd class to use
+        :param store_manager: Proxy object for remtote stores
```
Comment (mbarnes) [inline src/commissaire/transport/ansibleapi.py]: "remtote"

File: src/commissaire/handlers/util.py [hunk 4]
```diff
@@ -57,6 +57,29 @@ def etcd_cluster_exists(name):
...
 
 
+def cluster_for_host(address, store_manager=None):
+    """
+    Checks to see if the the host is part of a cluster. KeyError is raised
+    if the host is not part of a cluster.
+
+    :param name: Name of a cluster
+    :type name: str
+    :param store_manager: Proxy object for remtote stores
```
Comment (mbarnes) [inline src/commissaire/handlers/util.py]: "remtote"

Also, I'm not sure I like the `store_manager` param being optional.  Might just be a personal preference but I don't like util functions fetching global resources for themselves.

**Comment by mbarnes** [comment]
Looks good, just some nits.

File: src/commissaire/jobs/investigator.py (hunk 1 above)
Comment (ashcrow) [inline src/commissaire/jobs/investigator.py]: Will fix.

File: src/commissaire/jobs/investigator.py (hunk 2 above)
Comment (ashcrow) [inline src/commissaire/jobs/investigator.py]: Will fix.

File: src/commissaire/handlers/util.py (hunk 4 above)
Comment (ashcrow) [inline src/commissaire/handlers/util.py]: I'm good with removing the optional status. When you say "fetching global" do you mean as in using the bus or as in getting a global variable passed to it?

File: src/commissaire/transport/ansibleapi.py (hunk 3 above)
Comment (ashcrow) [inline src/commissaire/transport/ansibleapi.py]: Will fix.

**Comment by ashcrow** [comment]
Updated. One question added around the "global" nit.

File: src/commissaire/handlers/util.py (hunk 4 above)
Comment (mbarnes) [inline src/commissaire/handlers/util.py]: In this case I meant using the bus.

File: src/commissaire/handlers/util.py [hunk 5]
```diff
@@ -64,7 +64,7 @@ def cluster_for_host(address, store_manager=None):
 
     :param name: Name of a cluster
     :type name: str
-    :param store_manager: Proxy object for remtote stores
+    :param store_manager: Remote object for remtote stores
```
Comment (mbarnes) [inline src/commissaire/handlers/util.py]: Gah, sorry, I was trying to point out a typo.  "Proxy object" is fine, but keep reading.  :smile:

**Comment by ashcrow** [comment]
:arrow_up:

**Comment by mbarnes** [comment]
@rh-atomic-bot r+

**Comment by ashcrow** [comment]
Restarting the test. Looks like a flake.

**Comment by ashcrow** [comment]
@rh-atomic-bot try

**Comment by mbarnes** [comment]
Homu seems stuck today so I merged it myself.

**Pull request closed without merging**